src/Evolution/Systems/ScalarAdvection/Initialize.hpp

```diff
@@ -0,0 +1,67 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/Initialization/DiscontinuousGalerkin.hpp"
+#include "Evolution/Initialization/Evolution.hpp"
+#include "Evolution/Systems/ScalarAdvection/System.hpp"
+#include "Evolution/Systems/ScalarAdvection/Tags.hpp"
+#include "Evolution/Systems/ScalarAdvection/VelocityField.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "ParallelAlgorithms/Initialization/MutateAssign.hpp"
+
+namespace ScalarAdvection {
+namespace Actions {
+/*!
+ * \ingroup InitializationGroup
+ * \brief Initialize items related to the ScalarAdvection system
+ *
+ * Add advection velocity field to the evolution databox
+ *
+ * DataBox changes:
+ *  - Adds:
+ *    `ScalarAdvection::Tags::VelocityField`
+ *  - Removes: nothing
+ *  - Modifies: nothing
+ *
+ */
+
+template <size_t Dim>
+struct InitializeVelocityField {
```

Comment:
You don't need this action, you can just use `Initialization::Actions::AddComputeTags<tmpl::list<ScalarAdvection::Tags::VelocityFieldCompute<Dim>>>` 

and if you needed it, you need to write a test for it :)

Reply:
Writing a test for this action seems to be a bit easier solution for now, but I agree that your `AddComputeTags` method should work as well.

When using AddComputeTags, how should I retrieve domain coordinates and pass it to VelocityFieldCompute? The main reason I used initialization action was I couldn't figure out how to do that. (related with the `argument_tags` problem I commented above)

Reply:
Well if you make `VelocityFieldCompute` an actual compute tag (it's current just a function inside a class), then all you need to do is have `Initialization::Actions::AddComputeTags<VelocityFieldCompute>` in your initialization actions list in the executable. I'm not sure how you can get simpler than that :)

Reply:
I see. So if I set `domain::Tags::Coordinates` as one of the `argument_tags` of a compute tag and add it to the initialization action, the code autonomously recognizes that and plug in coordinates of each domain (without any need to explicitly retrieving mesh and coordinates as I did in `Initialize.hpp`) ?

Reply:
yes, that's right